### ISSUE TITLE
UseEffect function part 2

### ISSUE THREAD

**Issue by amykuner** [fac22/week6-MAD#33]
Correct me if I'm wrong, but you use this same function in a useEffect twice in your code. It could be good to bring this function out into another file to call in your useEffects (DRY code and all that jazz)

https://github.com/fac22/week6-MAD/blob/27f86c4eefc27dee380ec43103e17730c7037a19/pages/products/%5Bname%5D.js#L30

**Comment by oliverjam** [comment]
Yeah it seems like the only purpose of the `useEffect` is to initialise the state defined in `_app.js`. In that case you may as well move the effect up to there, so it's defined in one place and each page will just have access to the state directly.

On that note there is a simpler way to initialise complex state like this: pass an "initialiser" function to `useState`. This is for when you need to compute the initial value, but ensure that initialisation only happens once.

https://reactjs.org/docs/hooks-reference.html#lazy-initial-state

```js
// pages/_app.js

function MyApp() {
  const [basket, setBasket] = React.useState(() => {
    // get anything you need out of localStorage here
    return stuff;
  });
  return ...
}
```
